packages/emitter-framework/src/typescript/components/interface-member.tsx

```diff
@@ -8,39 +9,50 @@ import { TypeExpression } from "./type-expression.js";
 export interface InterfaceMemberProps {
   type: ModelProperty | Operation;
   optional?: boolean;
+  refkey?: ay.Refkey;
 }
 
-export function InterfaceMember({ type, optional }: InterfaceMemberProps) {
-  const namer = useTSNamePolicy();
-  const name = namer.getName(type.name, "object-member-getter");
+export function InterfaceMember(props: InterfaceMemberProps) {
+  const namer = ts.useTSNamePolicy();
+  const name = namer.getName(props.type.name, "object-member-getter");
 
-  if ($.modelProperty.is(type)) {
-    const optionality = optional === true || type.optional === true ? "?" : "";
-
-    if (isNeverType(type.type)) {
+  if ($.modelProperty.is(props.type)) {
+    if (isNeverType(props.type.type)) {
       return null;
     }
 
-    let unpackedType = type.type;
-    const part = getHttpPart($.program, type.type);
+    let unpackedType = props.type.type;
+    const part = getHttpPart($.program, props.type.type);
     if (part) {
       unpackedType = part.type;
     }
 
     return (
-      <>
-        "{name}"{optionality}: <TypeExpression type={unpackedType} />;
-      </>
+      <ts.InterfaceMember
+        name={name}
+        optional={props.optional === true || props.type.optional === true}
+        type={<TypeExpression type={unpackedType} />}
+        refkey={ay.refkey(props.type)}
+        readonly={Boolean(props.type.decorators?.find((d) => d.decorator.name === "@readonly"))}
+      />
     );
   }
 
-  if ($.operation.is(type)) {
-    const returnType = <TypeExpression type={type.returnType} />;
-    const params = <FunctionDeclaration.Parameters type={type.parameters} />;
+  if ($.operation.is(props.type)) {
+    const returnType = <TypeExpression type={props.type.returnType} />;
+    const params = <FunctionDeclaration.Parameters type={props.type.parameters} />;
+
     return (
-      <>
-        {name}({params}): {returnType};
-      </>
+      <ts.InterfaceMember
+        name={name}
+        type={
+          <>
+            ({params}): {returnType}
```

Comment:
```suggestion
            ({params}) => {returnType}
```

I think this might have been incorrect before as well

Reply:
agree

Reply:
Actually nevermind - I _think_ a test already exists I just have to unskip it 👍 https://github.com/microsoft/typespec/blob/main/packages/emitter-framework/test/typescript/components/interface-declaration.test.tsx#L511-L519